lib/testcase/TestCaseMediaTypes.ts

```diff
@@ -0,0 +1,48 @@
+
+// The media mappings for each parsing test suite
+
+export default {
+  // RDF/XML test suite
+  'http://www.w3.org/ns/rdftest#TestXMLEval': "application/xml",
+  'http://www.w3.org/ns/rdftest#TestXMLNegativeSyntax': "application/xml",
+
+  // N-Triples test suite
+  'http://www.w3.org/ns/rdftest#TestNTriplesPositiveSyntax': "application/n-triples",
+  'http://www.w3.org/ns/rdftest#TestNTriplesNegativeSyntax': "application/n-triples",
+
+  // N-Quads test suite
+  'http://www.w3.org/ns/rdftest#TestNQuadsPositiveSyntax': "application/n-quads",
+  'http://www.w3.org/ns/rdftest#TestNQuadsNegativeSyntax': "application/n-quads",
+
+  // Turtle test suite
+  'http://www.w3.org/ns/rdftest#TestTurtleEval': "text/turtle",
+  'http://www.w3.org/ns/rdftest#TestTurtlePositiveSyntax': "text/turtle",
+  'http://www.w3.org/ns/rdftest#TestTurtleNegativeSyntax': "text/turtle",
+  'http://www.w3.org/ns/rdftest#TestTurtleNegativeEval': "text/turtle",
+
+  // TriG test suite
+  'http://www.w3.org/ns/rdftest#TestTrigEval': "application/trig",
+  'http://www.w3.org/ns/rdftest#TestTrigPositiveSyntax': "application/trig",
+  'http://www.w3.org/ns/rdftest#TestTrigNegativeSyntax': "application/trig",
+  'http://www.w3.org/ns/rdftest#TestTrigNegativeEval': "application/trig",
+
+  // JSON-LD test suite
+  'https://w3c.github.io/json-ld-api/tests/vocab#ToRDFTest https://w3c.github.io/json-ld-api/tests/vocab#PositiveEvaluationTest': "application/ld+json",
+  'https://w3c.github.io/json-ld-api/tests/vocab#ToRDFTest https://w3c.github.io/json-ld-api/tests/vocab#PositiveSyntaxTest': "application/ld+json",
+  'https://w3c.github.io/json-ld-api/tests/vocab#ToRDFTest https://w3c.github.io/json-ld-api/tests/vocab#NegativeEvaluationTest': "application/ld+json",
+  'https://w3c.github.io/json-ld-api/tests/vocab#FromRDFTest https://w3c.github.io/json-ld-api/tests/vocab#PositiveEvaluationTest': "application/ld+json",
+  'https://w3c.github.io/json-ld-api/tests/vocab#FromRDFTest https://w3c.github.io/json-ld-api/tests/vocab#NegativeEvaluationTest': "application/ld+json",
```

Comment:
This is the only thing I am now unsure about - it arises from the keys in the `TestCaseHandlers.ts` file.

Reply:
I'm not sure if this approach can capture the media types for different test types.

For example, in the JSON-LD test suite, some tests will have an HTML content type, which depends on the presence of another test type (for example https://w3c.github.io/json-ld-api/tests/html-manifest.html#tex01).
For these tests, the test suite will even pass a custom `contentType` parameter to the parser.

Would it instead make sense to only capture the edge-cases in this file (Turtle/N3), and not do anything in all other cases?

Reply:
Ok!

Reply:
> For example, in the JSON-LD test suite, some tests will have an HTML content type, which depends on the presence of another test type (for example https://w3c.github.io/json-ld-api/tests/html-manifest.html#tex01).
For these tests, the test suite will even pass a custom contentType parameter to the parser.


... I've also just discovered that the current implementation causes no spec tests to be run when testing against the jsonld-streaming-parser so we need to fix this as well. Debugging now.